whoc/interfaces/hercules_hybrid_actuator_disk_interface.py

```diff
@@ -0,0 +1,102 @@
+# Copyright 2021 NREL
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# See https://nrel.github.io/wind-hybrid-open-controller for documentation
+
+# How will we handle other things here? May need to have a wind farm
+# version, an electrolyzer version, etc...
+from whoc.controllers.wind_farm_power_tracking_controller import POWER_SETPOINT_DEFAULT
+from whoc.interfaces.interface_base import InterfaceBase
+
+
+class HerculesHybridADInterface(InterfaceBase):
+    def __init__(self, hercules_dict):
+        super().__init__()
+
+        self.dt = hercules_dict["dt"]
+        self.n_turbines = hercules_dict["controller"]["num_turbines"]
+        self.turbines = range(self.n_turbines)
+
+        # Grab name of wind, solar, and battery (assumes there is only one!)
+        self.wind_name = list(hercules_dict["hercules_comms"]["amr_wind"].keys())[0]
+        py_sims = list(hercules_dict["py_sims"].keys())
+        self.solar_name = [ps for ps in py_sims if "solar" in ps][0]
+        self.battery_name = [ps for ps in py_sims if "battery" in ps][0]
+
+    def get_measurements(self, hercules_dict):
+        turbine_powers = (
+            hercules_dict["hercules_comms"]["amr_wind"][self.wind_name]["turbine_powers"]
+        )
+        time = hercules_dict["time"]
+
+        if ("external_signals" in hercules_dict
+            and "plant_power_reference" in hercules_dict["external_signals"]):
+            plant_power_reference = hercules_dict["external_signals"]["wind_power_reference"]
```

Comment:
Does this mean that the plant power reference needs to be defined as "wind_power_reference" always?

Reply:
@genevievestarke good catch!! That was supposed to be `"plant_power_reference"` (copy and paste error). I've now fixed this